src/components/snippetpanel.ts

```diff
@@ -0,0 +1,230 @@
+                if (symbol.svg === undefined) {
+                    mathSymbolPromises.push(
+                        new Promise((resolve, reject) => {
+                            this.mathJax
+                                .typeset({
+                                    math: symbol.source,
+                                    format: 'TeX',
+                                    svgNode: true
+                                })
+                                .then(
+                                    (data: {
+                                        height: string;
+                                        speakText: string;
+                                        style: string;
+                                        svgNode: SVGSVGElement;
+                                        width: string;
+                                    }) => {
+                                        let svg = data.svgNode.outerHTML
+                                        svg = svg.replace(
+                                            /<title([^>]*)>(.*)<\/title>/,
+                                            `<title$1>${symbol.name.toLocaleUpperCase()}.${
+                                                symbol.keywords ? ' Keywords: ' + symbol.keywords : ''
+                                            }</title>`
+                                        )
+                                        if (symbol.shrink) {
+                                            svg = svg.replace(/^<svg/, '<svg class="shrink"')
+                                        }
+                                        symbol.svg = svg
+                                        resolve()
+                                    }
+                                )
+                                .catch(reject)
+                        })
+                    )
+                }
+            }
+        }
+        Promise.all(mathSymbolPromises).finally(() => {
+            if (mathSymbolPromises.length > 0) {
+                writeFileSync(snipetsFile, JSON.stringify(snippets, undefined, 4))
+                vscode.window.showInformationMessage(
+                    `LaTeX-Workshop: ${mathSymbolPromises.length} symbols rendered and cached`
+                )
+            }
+            for (const category in snippets.mathSymbols) {
+                for (let i = 0; i < snippets.mathSymbols[category].length; i++) {
+                    const symbol = snippets.mathSymbols[category][i]
+                    symbol.category = category
+                    if (symbol.keywords === undefined) {
+                        symbol.keywords = ''
+                    }
+                    // @ts-ignore
+                    this.mathSymbols.push(symbol)
+                }
+            }
+        })
+    }
+
+    private async initialisePanel() {
+        if (this.panel === undefined) {
+            return
+        }
+
+        this.mathSymbols.forEach(async mathSymbol => {
+            // @ts-ignore
```

Comment:
Without the `@ts-ignore`, on line 204 I get the error (reffering to `this.panel`)
```
Object is possibly 'undefined'.
```
However this is impossible since if `this.panel` is `undefined` the function would have exited on line 199.
Hence tsc is incorrect.

Reply:
Don't use @ts-ignore. Just fix it. There is no exception. I don't want to discuss about this.

Reply:
The benefit of using a type system is that it allows us not to think about these things.

Reply:
Well sure, and I appreciate the benefits of doing a type system. I'm not against removing the `@ts-ignore` nor do I want to argue about it — I just can't see an elegant way to resolve this, and the error it's raising being … incorrect doesn't help me find a good solution.

If you've got a good solution that would be very helpful.

Reply:
How about this patch? 

```diff
diff --git a/src/components/snippetpanel.ts b/src/components/snippetpanel.ts
index 55d0af2..4d91a9d 100644
--- a/src/components/snippetpanel.ts
+++ b/src/components/snippetpanel.ts
@@ -200,7 +200,9 @@ export class SnippetPanel {
         }
 
         this.mathSymbols.forEach(async mathSymbol => {
-            // @ts-ignore
+            if (this.panel === undefined) {
+                return
+            }
             this.panel.webview.postMessage({
                 type: 'mathSymbol',
                 ...mathSymbol
```

Reply:
Done in b6f923ea51b9f786726df114c3e809f0071e9763